website/docs/reference/resource-configs/snowflake-configs.md

```diff
@@ -303,9 +303,9 @@ models:
 
 ## Temporary Tables
 
-Beginning in dbt version 1.3, incremental table merges for Snowflake utilize a `view` rather than a `temporary table`. The reasoning was to avoid the database write step that a temporary table would initiate and save compile time. 
+Beginning in dbt version 1.3, incremental table merges for Snowflake may utilize a `view` rather than a `temporary table`. The reasoning was to avoid the database write step that a temporary table would initiate and save compile time. 
 
-However, many situations remain where a temporary table would achieve results faster. dbt v1.4 adds the `tmp_relation_type` configuration to leverage temporary tables for incremental builds. This is defined as part of the model configuration.
+However, many situations remain where a temporary table would achieve results faster and is used. dbt v1.4 adds the `tmp_relation_type` configuration to leverage temporary tables for incremental builds. This is defined as part of the model configuration. However, you may see an error if an incremental model has `delete+insert` strategy with a unique key declared. In this instance the model must be configured with a `tmp_relation_type: table`.
```

Comment:
```suggestion
However, some situations remain where a temporary table would achieve results faster or more safely. dbt v1.4 adds the `tmp_relation_type` configuration to allow you to opt in to temporary tables for incremental builds. This is defined as part of the model configuration. 

To guarantee accuracy, an incremental model using the `delete+insert` strategy with a `unique_key` defined automatically defaults to using a temporary table; trying to change this to a view will result in an error.
```

I reworked this a bit, because I don't think it's correct to say that the model _needs to be_ explicitly configured. If I'm [interpreting the linked code](https://github.com/dbt-labs/dbt-snowflake/blob/ed6ed81ec2f8b409b29dafe11ed47a75b477ffec/dbt/include/snowflake/macros/materializations/incremental.sql#L35-L57) from your Slack message correctly, it looks like it can either be left unset _or_ `table` must be set (which is redundant since that will happen anyway).

Reply:
This is correct, but I also am trying to add the detail that some configuration settings are forbidden and raise the compiler error [here](https://github.com/dbt-labs/dbt-snowflake/blob/ed6ed81ec2f8b409b29dafe11ed47a75b477ffec/dbt/include/snowflake/macros/materializations/incremental.sql#L36). 

I was able to generate the compilation error in my environment with the following configurations: 
```
{{
    config(
        materialized='incremental',
        unique_key ='zendesk_user_id',
        incremental_strategy='delete+insert',
        tmp_relation_type='view'
    )
}}
```
<img width="832" alt="image" src="https://github.com/dbt-labs/docs.getdbt.com/assets/8473398/8ba7e615-692f-4ec7-bbaa-1cd4a388b947">

Reply:
My main goal of the PR is to clarify the wording a bit since when I first read it I was surprised my environment was still using tmp tables, but in fact, tables are still used quite often based on the configurations. 

Thanks for your feedback!